sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultBackupClient.cs

```diff
@@ -126,26 +126,26 @@ public virtual BackupOperation StartBackup(Uri blobStorageUri, string sasToken,
         /// <summary>
         /// Initiates a full restore of the Key Vault.
         /// </summary>
-        /// <param name="backupFolderUri">
+        /// <param name="folderUri">
         /// The <see cref="Uri"/> for the blob storage resource, including the path to the blob container where the backup resides.
         /// This would be the exact value that is returned as the result of a <see cref="BackupOperation"/>.
         /// An example Uri may look like the following: https://contoso.blob.core.windows.net/backup/mhsm-contoso-2020090117323313.
         /// </param>
         /// <param name="sasToken">A Shared Access Signature (SAS) token to authorize access to the blob.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
-        /// <exception cref="ArgumentNullException"><paramref name="backupFolderUri"/> or <paramref name="sasToken"/> is null.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="folderUri"/> or <paramref name="sasToken"/> is null.</exception>
         /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
         /// <returns>A <see cref="RestoreOperation"/> to wait on this long-running operation.</returns>
-        public virtual async Task<RestoreOperation> StartRestoreAsync(Uri backupFolderUri, string sasToken, CancellationToken cancellationToken = default)
+        public virtual async Task<RestoreOperation> StartRestoreAsync(Uri folderUri, string sasToken, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(KeyVaultBackupClient)}.{nameof(StartRestore)}");
             scope.Start();
             try
             {
                 // Get the folder name from the backupBlobUri returned from a previous BackupOperation
-                string[] uriSegments = backupFolderUri.Segments;
+                string[] uriSegments = folderUri.Segments;
                 string folderName = uriSegments[uriSegments.Length - 1];
```

Comment:
I don't think this will be right, as we discussed in the meeting. If nested folders are supported, someone passing "https://storage/container/a/b" would mean the folder is "a/b" and you only grabbed "b". You should add tests for this case.

Reply:
...and be sure to review the recorded session JSON. It may seem to work, but validate that the result is actually expected (i.e. did the backup end up in /container/a/b)?

Reply:
I had planned to address that in a different PR

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/17332

Reply:
#17332